internal/condition/eval/eval.go

```diff
@@ -2,13 +2,27 @@ package eval
 
 import (
 	"fmt"
+	"time"
 
 	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/openfga/openfga/internal/build"
 	"github.com/openfga/openfga/internal/condition"
 	"github.com/openfga/openfga/pkg/typesystem"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
+var conditionEvaluationCostHistogram = promauto.NewHistogram(prometheus.HistogramOpts{
+	Namespace:                       build.ProjectName,
+	Name:                            "condition_evaluation_cost",
+	Help:                            "A histogram of the CEL evaluation cost of a Condition in a Relationship Tuple",
+	Buckets:                         []float64{1, 5, 15, 20, 35, 50, 75, 90, 100},
```

Comment:
- [ ] todo: the upper bucket here should be conditionally defined by the max evaluation cost variable. Ideally it shouldn't be hard coded.